index.js

```diff
@@ -1,73 +1,81 @@
-var async = require('async');
+module.exports = Sevnup;
 
-var VNodeStore = require('./lib/vnode-store.js');
+var _ = require('lodash');
+var crypto = require('crypto');
+var async = require('async');
 
-//TODO (joseph@): Config.
-var TOTAL_VNODES = 14;
-var MAX_PARALLEL_TASKS = 5;
+var DEFAULT_TOTAL_VNODES = 1024;
+var MAX_PARALLEL_TASKS = 10;
 
 /**
- * Constructor, takes all optional persistence function overrides but expects none.
- * @constructor
- * @param {function} loadVNKeysFromStorage A method that takes a vnode and
- *     the list of keys it owns, presumably recovered from a datastore.
- *  @param {function} persistKeyToVNode Given a key and a VNode, this function
- *      adds the relation to the datastore.
- *  @param {function} persistRemoveKeyFromVNode The inverse of
- *      persistKeyToVNode, removes a key relation to a VNode in the store.
- *  @param {function} recoverKey The function to run on each key that is
- *      recovered. Takes 'done' callback.
- *  @param {function} releaseKey The function called when you release your
- *      ownership of a key, for example if another node now owns it.  Cleanup.
- *      Also takes a 'done' callback.
+ * Params are:
+ *   hashRing
+ *   loadVNodeKeysFromStorage
+ *   persistKeyToVNode
+ *   persistRemoveKeyFromVNode
+ *   recoverKey
+ *   releaseKey
+ *   totalVNodes
+ *   logger
  */
-function Sevnup(persistenceService, recoverKey, releaseKey) {
-    var allVNodes = [];
-    for (var i=0; i<TOTAL_VNODES; i++) {
-        allVNodes.push(i);
-    }
-    this.allVNodes = allVNodes;
-    this.vnodeStore = new VNodeStore(persistenceService, recoverKey, releaseKey);
+function Sevnup(params) {
+    this.hashRing = params.hashRing;
+    this.hashRingLookup = this.hashRing.lookup.bind(this.hashRing);
+    this.loadVNodeKeysFromStorage = params.loadVNodeKeysFromStorage;
+    this.persistKeyToVNode = params.persistKeyToVNode;
+    this.persistRemoveKeyFromVNode = params.persistRemoveKeyFromVNode;
+    this.recoverKeyCallback = params.recoverKey;
+    this.releaseKeyCallback = params.releaseKey;
+    this.totalVNodes = params.totalVNodes || DEFAULT_TOTAL_VNODES;
```

Comment:
removing the persistence object makes it so you can't swap in specific persistence models, like psql or riak.  needs to be there